encodings/alp/src/alp/mod.rs

```diff
@@ -212,12 +213,10 @@ fn encode_chunk_unchecked<T: ALPFloat>(
 
     // encode the chunk, counting the number of patches
     let mut chunk_patch_count = 0;
-    encoded_output.extend(chunk.iter().map(|v| {
-        let encoded = unsafe { T::encode_single_unchecked(*v, exp) };
-        let decoded = T::decode_single(encoded, exp);
-        let neq = (decoded != *v) as usize;
-        chunk_patch_count += neq;
-        encoded
+    encoded_output.extend(chunk.iter().copied().map(|v| {
+        let encoded = T::encode_single(v, exp);
+        chunk_patch_count += if encoded.is_none() { 1 } else { 0 };
+        encoded.unwrap_or_else(|| T::MAX_INT.as_int())
```

Comment:
The original impl fills in the ENCODING_UPPER_LIMIT for all values that fail the roundtrip